source/features/add-delete-to-pr-files.js

```diff
@@ -0,0 +1,21 @@
+import select from 'select-dom';
+import {h} from 'dom-chef';
+import * as pageDetect from '../libs/page-detect';
+
+const repoURL = pageDetect.getRepoURL();
+
+export default function () {
+	const findBranch = select('.head-ref').innerText;
+	const branchName = findBranch.includes(':') ? findBranch.split(':')[1] : findBranch;
+
+	const filesInfo = select.all('.file-info');
+	const generateURL = filesInfo.map(file => `/${repoURL}/delete/${branchName}/${file.children[1].title}`);
+
+	const fileActionContainers = select.all('div.BtnGroup').slice(1);
+	fileActionContainers.forEach((container, index) => {
+		container.prepend(<a
+			href={generateURL[index]}
+			className="btn btn-sm tooltipped tooltipped-s BtnGroup-item"
+			aria-label="Delete this file from the pull request">Delete</a>);
+	});
```

Comment:
Each "file info" is connected to a "file action container", so no need to select them separately and use indexes.

Find a container and then query both, something like:

```js
for (const file of select.all('.file-header')) {
	const fileName = select('.file-info a', file).title;
	const url = `/${repoURL}/delete/${branchName}/${fileName}`
	select('.BtnGroup', file).prepend(
		<a
			href={url}
			...>
			Delete
		</a>
	);
}
```

Also those extra `[1]` and `.slice(1)` were a code smell because the first one depended on the exact position of an element and the second one was the consequence of a selector that included unrelated `BtnGroup`s